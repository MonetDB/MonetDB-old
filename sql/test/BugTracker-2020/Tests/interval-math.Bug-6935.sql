SELECT INTERVAL '1' HOUR / 2, INTERVAL '1' HOUR * 1000 / 2000; --all output 1800.000
SELECT INTERVAL '1' HOUR / 2.0; -- 1800.000
SELECT INTERVAL '1' HOUR * 1000.0 / 2000.0; --1800.000
SELECT INTERVAL '1' HOUR * 1000 / 1800000; -- 2.000
SELECT INTERVAL '1' HOUR * CAST(1000 AS DOUBLE); --cannot multiply interval with floating-point
SELECT INTERVAL '1' MONTH * 1.2; -- 1
SELECT INTERVAL '1' MONTH / 2.0; -- 0
SELECT INTERVAL '1' MONTH / 1.0; -- 1
SELECT INTERVAL '1' SECOND * 1.2; --1.200
SELECT INTERVAL '1' HOUR / INTERVAL '1800' SECOND; --error on typing branch, cannot divide intervals
