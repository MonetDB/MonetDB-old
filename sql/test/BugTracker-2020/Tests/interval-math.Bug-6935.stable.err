stderr of test 'interval-math.Bug-6935` in directory 'sql/test/BugTracker-2020` itself:


# 10:13:31 >  
# 10:13:31 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-80282" "--port=37020"
# 10:13:31 >  

MAPI  = (monetdb) /var/tmp/mtest-465592/.s.monetdb.32432
QUERY = SELECT INTERVAL '1' HOUR * CAST(1000 AS DOUBLE); --cannot multiply interval with floating-point
ERROR = !overflow in conversion of 1000 to bte.
CODE  = 22003
MAPI  = (monetdb) /var/tmp/mtest-237353/.s.monetdb.30721
QUERY = SELECT INTERVAL '1' HOUR / INTERVAL '1800' SECOND; --error on typing branch, cannot divide intervals
ERROR = !SELECT: no such binary operator 'sql_div(sec_interval,sec_interval)'
CODE  = 42000

# 12:56:20 >  
# 12:56:20 >  "Done."
# 12:56:20 >  

