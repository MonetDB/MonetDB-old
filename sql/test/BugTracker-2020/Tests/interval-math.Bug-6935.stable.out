stdout of test 'interval-math.Bug-6935` in directory 'sql/test/BugTracker-2020` itself:


# 10:13:31 >  
# 10:13:31 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-80282" "--port=37020"
# 10:13:31 >  

#SELECT INTERVAL '1' HOUR / 2, INTERVAL '1' HOUR * 1000 / 2000; --all output 1800.000
% .%2,	.%3 # table_name
% %2,	%3 # name
% sec_interval,	sec_interval # type
% 8,	8 # length
[ 1800.000,	1800.000	]
#SELECT INTERVAL '1' HOUR / 2.0; -- 1800.000
% .%2 # table_name
% %2 # name
% sec_interval # type
% 8 # length
[ 1800.000	]
#SELECT INTERVAL '1' HOUR * 1000.0 / 2000.0;
% .%2 # table_name
% %2 # name
% sec_interval # type
% 8 # length
[ 1800.000	]
#SELECT INTERVAL '1' HOUR * 1000 / 1800000; -- 2.000
% .%2 # table_name
% %2 # name
% sec_interval # type
% 5 # length
[ 2.000	]
#SELECT INTERVAL '1' MONTH * 1.2;
% .%2 # table_name
% %2 # name
% month_interval # type
% 1 # length
[ 1	]
#SELECT INTERVAL '1' SECOND * 1.2; --1.200
% .%2 # table_name
% %2 # name
% sec_interval # type
% 5 # length
[ 1.200	]

# 10:13:31 >  
# 10:13:31 >  "Done."
# 10:13:31 >  

